pysteps/io/importers.py

```diff
@@ -135,6 +135,119 @@
     PYPROJ_IMPORTED = False
 
 
+def import_crri_eu(filename, **kwargs):
+    """Import a NetCDF radar rainfall product from the CRRI SAF.
+
+    Parameters
+    ----------
+    filename : str
+        Name of the file to import.
+
+    Returns
+    -------
+    out : tuple
+        A three-element tuple containing the rainfall field in mm/h imported
+        from the CRRI SAF netcdf, the quality field and the metadata. The
+        quality field is currently set to None.
+
+    """
+    if not NETCDF4_IMPORTED:
+        raise MissingOptionalDependency(
+            "netCDF4 package is required to import CRRI SAF products "
+            "but it is not installed"
+        )
+
+    R = _import_crri_eu_data(filename)
+
+    geodata = _import_crri_eu_geodata(filename)
+    metadata = geodata
+
+    # TODO(import_crri_eu): Add missing georeferencing data.
+
+    metadata["transform"] = None
+    metadata["zerovalue"] = np.nanmin(R)
+    if np.any(np.isfinite(R)):
+        metadata["threshold"] = np.nanmin(R[R > np.nanmin(R)])
+    else:
+        metadata["threshold"] = np.nan
+
+    # cut a window, to be defined here and only here
+    if True:
```

Comment:
What is the motivation to include the domain clipping in the importer?
Personally I would prefer to keep this sort of data manipulation outside of the importers module. If this was motivated by memory limits, perhaps we could add a general option in the readers module, in order to apply some data processing operations (eg clipping, upscaling,..) while reading the data.

Besides, there is a utils method already implemented in pysteps to clip rasters based on geographical coordinates ([clip_domain()](https://pysteps.readthedocs.io/en/latest/generated/pysteps.utils.dimension.clip_domain.html#pysteps.utils.dimension.clip_domain)). Probably worth to have a look at it to see if it can already perform the operation you need.

Reply:
solved by 99d6b65